var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -20,78 +20,98 @@ class Dyninst(Package):
+
+    depends_on('cmake@3.0:', type='build', when='@10.0:')
+    depends_on('cmake@2.8:', type='build', when='@:9.99')
 
     patch('stat_dysect.patch', when='+stat_dysect')
     patch('stackanalysis_h.patch', when='@9.2.0')
 
-    # new version uses cmake
-    def install(self, spec, prefix):
-        if spec.satisfies('@:8.1'):
-            configure("--prefix=" + prefix)
-            make()
-            make("install")
-            return
-
-        libelf = spec['elf'].prefix
-        if spec.satisfies('@:9'):
-            libdwarf = spec['libdwarf'].prefix
-
-        with working_dir('spack-build', create=True):
-            args = ['..',
-                    '-DBoost_INCLUDE_DIR=%s'    % spec['boost'].prefix.include,
-                    '-DBoost_LIBRARY_DIR=%s'    % spec['boost'].prefix.lib,
-                    '-DBoost_NO_SYSTEM_PATHS=TRUE',
-                    '-DLIBELF_INCLUDE_DIR=%s'   % join_path(
-                        libelf.include, 'libelf'),
-                    '-DLIBELF_LIBRARIES=%s'     % join_path(
-                        libelf.lib, "libelf." + dso_suffix)]
-            if spec.satisfies('@:9'):
-                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libdwarf.include)
-                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
-                    libdwarf.lib, "libdwarf." + dso_suffix))
-            # For @develop + use elfutils libdw, libelf is an abstraction
-            # we are really using elfutils here
-            if spec.satisfies('@develop') or spec.satisfies('@10:'):
-                tbb = spec['tbb'].prefix
-                args.append('-DTBB_INCLUDE_DIRS=%s' % tbb.include)
-                args.append('-DTBB_LIBRARIES=%s'    % join_path(
-                    tbb.lib, "libtbb." + dso_suffix))
-                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libelf.include)
-                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
-                    libelf.lib, "libdw." + dso_suffix))
-            if spec.satisfies('arch=linux-redhat7-ppc64le'):
-                args.append('-Darch_ppc64_little_endian=1')
-            args += std_cmake_args
-            cmake(*args)
-            make()
-            make("install")
-
-    @when('@:8.1')
-    def install(self, spec, prefix):
-        configure("--prefix=" + prefix)
-        make()
-        make("install")
+    # Versions 9.3.x used cotire, but have no knob to turn it off.
+    # Cotire has no real use for one-time builds and can break
+    # parallel builds with both static and shared libs.
+    @when('@9.3.0:9.3.99')
+    def patch(self):
+        filter_file('USE_COTIRE true', 'USE_COTIRE false',
+                    'cmake/shared.cmake')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Elf -- the directory containing libelf.h, either
+        # prefix/include (elfutils), or prefix/include/libelf
+        # (libelf).
+        elf = spec['elf'].prefix
+        if spec.satisfies('@9.3.0:'):
```

Comment:
IMO it would be clearer to base this off the elf version being used: `if 'elf@1' in spec`

FWIW, regarding

>  It looks like spec['elf'].headers returns a list of .h files, but what I need is the directory containing them. I could take dirname() of one element in the list, but that seems a bit clunky.

you can say `spec['elf'].headers.directories` to retrieve the directories with `.h` files in them.